packages/mui-material/src/Chip/Chip.js

```diff
@@ -156,141 +146,197 @@ const ChipRoot = styled('div', {
-          '&:hover, &:active': {
-            color: (theme.vars || theme).palette[ownerState.color].contrastText,
-          },
-        }),
       },
-      ...(ownerState.size === 'small' && {
-        height: 24,
-      }),
-      ...(ownerState.color !== 'default' && {
-        backgroundColor: (theme.vars || theme).palette[ownerState.color].main,
-        color: (theme.vars || theme).palette[ownerState.color].contrastText,
-      }),
-      ...(ownerState.onDelete && {
-        [`&.${chipClasses.focusVisible}`]: {
-          backgroundColor: theme.vars
-            ? `rgba(${theme.vars.palette.action.selectedChannel} / calc(${theme.vars.palette.action.selectedOpacity} + ${theme.vars.palette.action.focusOpacity}))`
-            : alpha(
-                theme.palette.action.selected,
-                theme.palette.action.selectedOpacity + theme.palette.action.focusOpacity,
-              ),
+      variants: [
+        {
+          props: { size: 'small' },
+          style: {
+            height: 24,
+            [`& .${chipClasses.icon}`]: {
+              fontSize: 18,
+              marginLeft: 4,
+              marginRight: -4,
+            },
+            [`& .${chipClasses.deleteIcon}`]: {
+              fontSize: 16,
+              marginRight: 4,
+              marginLeft: -4,
+            },
+          },
         },
-      }),
-      ...(ownerState.onDelete &&
-        ownerState.color !== 'default' && {
-          [`&.${chipClasses.focusVisible}`]: {
-            backgroundColor: (theme.vars || theme).palette[ownerState.color].dark,
+        ...Object.entries(theme.palette)
+          .filter(([, value]) => value.main && value.contrastText)
+          .map(([color]) => {
+            return {
+              props: { color },
+              style: {
+                backgroundColor: (theme.vars || theme).palette[color].main,
+                color: (theme.vars || theme).palette[color].contrastText,
+                [`& .${chipClasses.deleteIcon}`]: {
+                  color: theme.vars
+                    ? `rgba(${theme.vars.palette[color].contrastTextChannel} / 0.7)`
+                    : alpha(theme.palette[color].contrastText, 0.7),
+                  '&:hover, &:active': {
+                    color: (theme.vars || theme).palette[color].contrastText,
+                  },
+                },
+              },
+            };
+          }),
+        {
+          props: (props) => props.iconColor === props.color,
+          style: {
+            [`& .${chipClasses.icon}`]: {
+              color: theme.vars ? theme.vars.palette.Chip.defaultIconColor : textColor,
+            },
           },
-        }),
+        },
+        {
+          props: (props) => props.iconColor === props.color && props.color !== 'default',
+          style: {
+            [`& .${chipClasses.icon}`]: {
+              color: 'inherit',
+            },
+          },
+        },
+        {
+          props: { onDelete: true },
```

Comment:
```suggestion
          props: ({ ownerState }) => !!ownerState.onDelete,
```

Reply:
Yes. If I understand correctly, the `ownerState` key is [checked here](https://github.com/mui/material-ui/blob/next/packages/mui-system/src/createStyled/createStyled.js#L72) and has precedence over props. Because of this, and because `onDelete` is [overwritten here](https://github.com/mui/material-ui/blob/next/packages/mui-material/src/Chip/Chip.js#L407), then the `ownerState` value is the one used.

Same for `clickable`.

Reply:
Ah, I see. It's overwritten. Thanks for the link.